osu.Game/Overlays/Chat/ChatLine.cs

```diff
@@ -112,13 +119,13 @@ private void load(OverlayColourProvider? colourProvider, OsuConfigManager config
                             Colour = colourProvider?.Background1 ?? Colour4.White,
                             AlwaysPresent = true,
                         },
-                        new MessageSender(message.Sender)
+                        new MessageSender(message.Sender, usernameColouredDrawable)
```

Comment:
Dunno if just me but I really do find this flow of doing things quite shocking. `createUsername()` returns a `usernameDrawable` to display as the username, but *also* in the background populates the `usernameColouredDrawable` field (making the method impure, even though it *looks* pure). `usernameColouredDrawable` is then passed here to `MessageSender` via ctor, but `usernameDrawable` itself is set as the child of `messageSender`...? It's really quite comical.

All of the `createUsername()` stuff needs to moved out of here, firmly inside of `MessageSender`, if you ask me. Skips this entire... whatever this method of data passing is.

Reply:
Not a fan of the branch above, rationale being that `MessageSender` as written in there is not really a "message sender" anything more than a generic "hover handler". I generally prefer single-use, concrete, specialised components to weird, overly generic, abstract thingies.